packages/plugin/webpack/src/WebpackConfig.ts

```diff
@@ -273,4 +280,121 @@ export default class WebpackConfigGenerator {
+    };
+    for (const entry of entryPoints) {
+      if (entry.nodeIntegration ?? this.pluginConfig.renderer.nodeIntegration) {
+        // Has node
+        if (isPreloadOnly(entry)) {
+          targets.electronPreload.push(entry);
+        } else {
+          targets.electronRenderer.push(entry);
+          if (isLocalWindow(entry) && entry.preload) {
+            targets.electronPreload.push(entry);
+          }
+        }
+      } else {
+        // No node
+        if (isPreloadOnly(entry)) {
+          targets.sandboxedPreload.push(entry);
+        } else {
+          targets.web.push(entry);
+          if (isLocalWindow(entry) && entry.preload) {
+            targets.sandboxedPreload.push(entry);
+          }
+        }
+      }
+    }
+    console.log('targets', targets);
+    return [
+      await this.buildRendererConfig(targets.web, RendererTarget.Web),
+      await this.buildRendererConfig(targets.electronRenderer, RendererTarget.ElectronRenderer),
+      await this.buildRendererConfig(targets.electronPreload, RendererTarget.ElectronPreload),
+      await this.buildRendererConfig(targets.sandboxedPreload, RendererTarget.SandboxedPreload),
+    ].filter(function isNotNull<T>(item: T | null): item is T {
+      return !!item;
+    });
+  }
+
+  buildRendererBaseConfig(target: RendererTarget): webpack.Configuration {
+    return {
+      target,
+      devtool: this.rendererSourceMapOption,
+      mode: this.mode,
+      output: {
+        path: path.resolve(this.webpackDir, 'renderer'),
+        filename: '[name]/index.js',
+        globalObject: 'self',
+        ...(this.isProd ? {} : { publicPath: '/' }),
+      },
+      node: {
+        __dirname: false,
+        __filename: false,
+      },
+      plugins: [new AssetRelocatorPatch(this.isProd, !!this.pluginConfig.renderer.nodeIntegration)],
+    };
+  }
+
+  async buildRendererConfig(entryPoints: WebpackPluginEntryPoint[], target: RendererTarget): Promise<Configuration | null> {
+    const entry: webpack.Entry = {};
+
+    if (entryPoints.length === 0) {
+      return null;
+    }
+    const baseConfig: webpack.Configuration = this.buildRendererBaseConfig(target);
+    const rendererConfig = await this.resolveConfig(this.pluginConfig.renderer.config);
+
+    if (target === RendererTarget.Web || target === RendererTarget.ElectronRenderer) {
+      for (const entryPoint of entryPoints) {
+        entry[entryPoint.name] = (entryPoint.prefixedEntries || []).concat([isPreloadOnly(entryPoint) ? entryPoint.preload.js : entryPoint.js]);
+      }
+      const output = {
+        path: path.resolve(this.webpackDir, 'renderer'),
+        filename: '[name]/index.js',
+        globalObject: 'self',
+        ...(this.isProd ? {} : { publicPath: '/' }),
+      };
+      const plugins: webpack.WebpackPluginInstance[] = [];
+      for (const entryPoint of entryPoints) {
+        if (isLocalWindow(entryPoint)) {
+          plugins.push(
+            new HtmlWebpackPlugin({
```

Comment:
is there a way we can avoid adding a plugin per entrypoint?

Reply:
No this is just how webpack works